save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt

```diff
@@ -75,19 +93,39 @@ class WarnPlugin(
         generalConfig: GeneralConfig?): List<TestResult> {
         val expectedWarnings: WarningMap = mutableMapOf()
         paths.forEach { path ->
+            val linesFile = fs.readLines(path)
             expectedWarnings.putAll(
-                fs.readLines(path)
-                    .mapNotNull {
-                        with(warnPluginConfig) {
-                            it.extractWarning(
-                                warningsInputPattern!!,
-                                path.name,
-                                lineCaptureGroup,
-                                columnCaptureGroup,
-                                messageCaptureGroup!!,
-                            )
+                if (warnPluginConfig.defaultLineMode!!) {
```

Comment:
Maybe, we can make it like this:
```kotlin
linesFile.mapIndexed { index, line ->
    if (warnPlugin.defaultLineMode!!) {
        // extraact warning with `plusLine`
    } else {
        // extractWarning as usual
    }
}
.filterNotNull()
```
Because having a large if expression, followed by a bunch of other calls with lambdas makes it really overwhelming. And, AFAIU, we don't really need to iterate twice when using `plusLine`

Reply:
Yes. Good idea:)
Fixed